### PR TITLE
debug pull to see how ci funtional tests work with SAML changes

### PR DESCRIPTION
**Pull request by arthurchan35**

### CODE REVIEW

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/action/SendAuthnRequestMVCResourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
+/**
```
Comment (liferay-appsec) [inline modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/action/SendAuthnRequestMVCResourceCommand.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/action/SendSamlSloRequestInfosMVCResourceCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
```
Comment (liferay-appsec) [inline modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/portlet/action/SendSamlSloRequestInfosMVCResourceCommand.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: portal-web/docroot/html/portal/saml/select_idp.jsp [hunk 3]
```diff
@@ -24,9 +24,11 @@ JSONObject samlSsoLoginContext = (JSONObject)request.getAttribute("SAML_SSO_LOGI
 JSONArray relevantIdpConnectionsJSONArray = samlSsoLoginContext.getJSONArray("relevantIdpConnections");
 %>
 
-<aui:form action='<%= PortalUtil.getPortalURL(request) + "/c/portal/login" %>' method="get" name="fm" style="padding: 1rem;">
-	<aui:input name="saveLastPath" type="hidden" value="<%= false %>" />
+<liferay-portlet:resourceURL id="/saml/send_authn_request" var="sendAuthnRequestURL" />
+
+<aui:form action="<%= sendAuthnRequestURL %>" id="fm" name="fm" style="padding: 1rem;">
 	<aui:input name="redirect" type="hidden" value="<%= redirect %>" />
```
Comment (liferay-appsec) [inline portal-web/docroot/html/portal/saml/select_idp.jsp:30]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/dxp/apps/saml/saml-web/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -46,6 +46,7 @@ incorrect-keystore-password=Incorrect keystore password.
 javax.portlet.title.1_WAR_samlportlet=SAML Keep Alive
```
Comment (liferay-appsec) [inline modules/dxp/apps/saml/saml-web/src/main/resources/content/Language.properties:46]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:reevaluate:1137401_5881

**Comment by arthurchan35** [comment]
ci:reevaluate:1137401_5881

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by stian-sigvartsen** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
Hey @stian-sigvartsen 
I have checked the new failures, and it's now different ones (they all fail for same reason).
Here is the detail:
https://testray.liferay.com/reports/production/logs/2021-05/test-1-12/test-portal-acceptance-pullrequest(master)/4609/functional-tomcat90-mysql57-jdk8/0/24/LocalFile.SAML_ViewSingleLogoutHTTPPostBindingInConsole/summary.html.gz

So basically if login from idp initiated SSO, the page wont jump back to sp page.
http://localhost:8080/c/portal/saml/redirect/sso?entityId=samlsp&RelayState=http://www.able.com:9080

The difficult part is that I cannot reproduce this problem locally.
I tested it locally and my page would jump back to sp page specified in RelayState.

**Comment by arthurchan35** [comment]
Looking at:
https://testray.liferay.com/reports/production/logs/2021-05/test-1-12/test-portal-acceptance-pullrequest(master)/4609/functional-tomcat90-mysql57-jdk8/0/24/jenkins-console.txt.gz

It says:
```
><saml2:AuthnContextClassRef>urn:oasis:names:tc:SAML:2.0:ac:classes:unspecified</saml2:AuthnContextClassRef></saml2:AuthnContext></saml2:AuthnStatement></saml2:Assertion></saml2p:Response> to http://www.able.com:9080/?p_p_id=com_liferay_saml_web_internal_portlet_SamlPortlet&p_p_lifecycle=1&_com_liferay_saml_web_internal_portlet_SamlPortlet_javax.portlet.action=/saml/assertion_consumer_service with binding urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST
     [exec] 2021-05-04 10:10:14.490 DEBUG [http-nio-8080-exec-9][VelocityEngineFactory:168] ResourceManager : found /templates/saml2-post-binding.vm with loader org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader
     [exec] 2021-05-04 10:10:14.493 DEBUG [http-nio-8080-exec-9][VelocityEngineFactory:168] ResourceManager : found /templates/add-html-head-content.vm with loader org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader
     [exec] 2021-05-04 10:10:14.494 DEBUG [http-nio-8080-exec-9][VelocityEngineFactory:168] ResourceManager : found /templates/add-html-body-content.vm with loader org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader
     [exec] 2021-05-04 10:10:14.574 WARN  [http-nio-8080-exec-9][code_jsp:173] {code="500", msg="", uri=/web/guest}
     [exec] javax.servlet.ServletException: java.lang.IllegalStateException: Unable to obtain Writer because OutputStream is already in use
     [exec] 	at com.liferay.portal.struts.PortalRequestProcessor._process(PortalRequestProcessor.java:420) ~[portal-impl.jar:?]
```

Is it possible related to https://issues.liferay.com/browse/LPS-13720

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:stop

**Comment by arthurchan35** [comment]
ci:test:saml:norebase

**Comment by arthurchan35** [comment]
ci:test:saml:norebase

**Comment by arthurchan35** [comment]
ci:test:saml:nocompile

**Comment by arthurchan35** [comment]
ci:test:saml:norebase

**Comment by arthurchan35** [comment]
ci:test:saml:norebase

**Pull request closed without merging**